lib/flowViewport/sceneIndex/fvpWireframeSelectionHighlightSceneIndex.cpp

```diff
@@ -50,6 +65,314 @@ const HdDataSourceLocator reprSelectorLocator(
 
 const HdDataSourceLocator primvarsOverrideWireframeColorLocator(
         HdPrimvarsSchema::GetDefaultLocator().Append(_primVarsTokens->overrideWireframeColor));
+
+const std::string selectionHighlightMirrorTag = "_SelectionHighlight";
+
+SdfPath _GetSelectionHighlightMirrorPathFromOriginal(const SdfPath& originalPath)
+{
+    return originalPath.ReplaceName(TfToken(originalPath.GetName() + selectionHighlightMirrorTag));
+}
+
+SdfPath _GetOriginalPathFromSelectionHighlightMirror(const SdfPath& mirrorPath)
+{
+    const std::string primName = mirrorPath.GetName();
+    return mirrorPath.ReplaceName(TfToken(primName.substr(0, primName.size() - selectionHighlightMirrorTag.size())));
+}
+
+// Computes the mask to use for an instancer's selection highlight mirror
+// based on the instancer's topology and its selections. This allows
+// highlighting only specific instances in the case of instance selections.
+VtBoolArray _GetSelectionHighlightMask(const HdInstancerTopologySchema& originalInstancerTopology, const HdSelectionsSchema& selections) 
+{
+    // Schema getters were made const in USD 24.05 (specifically Hydra API version 66).
+    // We work around this for previous versions by const casting.
+    VtBoolArray originalMask = 
+#if HD_API_VERSION < 66
+    const_cast<HdInstancerTopologySchema&>(originalInstancerTopology).GetMask()->GetTypedValue(0);
+#else
+    originalInstancerTopology.GetMask()->GetTypedValue(0);
+#endif
```

Comment:
Maybe factor this out in an unnamed namespace function?
```suggestion
inline HdInstancerTopologySchema& topologySchema(HdInstancerTopologySchema& t)
#if HD_API_VERSION < 66
     return const_cast<HdInstancerTopologySchema&>(t);
#else
    return t;
#endif
```

Reply:
I'm not sure I see what value this brings. If the method takes in a const reference, and does whatever treatment it needs to return a non-const reference, isn't that just a more complicated way of doing a flat-out `const_cast`? And if it takes in a non-const reference, then there would be no need to use `const_cast` in the first place? To me this preprocessor bit is mostly to confine the const_casting to the old Hydra versions, and use the new ones for the current & future (which would also allow us to remove the old API usage at a future point should we wish to do so)

Reply:
It was just a suggestion for readability and being closer to Don't Repeat Yourself.  You have the const_cast ifdef twice, and inside the ifdef you're repeating code.  No big deal though.